meta-resin-common/classes/binary-compress.bbclass

```diff
@@ -0,0 +1,28 @@
+# e.g.  COMPRESS_BINARY = "path/bin1 path/bin2 path/bin3"
+# Define what binaries that we find in the package tree will be compress
+# This variable's definition is MANDATORY if this class is used
+# NOTE: path is relative to PKGD directory
+COMPRESS_BINARY ?= ""
+
+DEPENDS_append = " bash upx"
+
+find_and_compress() {
+    #Sanity check
+    if [ -z ${COMPRESS_BINARY} ]; then
+        bbwarn "Binary compress class imported but COMPRESS_BINARY variable was found empty. Skipping compression."
+    else
+        #Find all executables once
+        pkg_dir=${PKGD}
+        execs=$(find $pkg_dir -executable -type f)
+        #Compress
+        for exec in $execs; do
+            bin=${exec#$pkg_dir}
+            binaries=${COMPRESS_BINARY}
+            if [ "${binaries#*$bin}" != "$binaries" ]; then
+                ${STAGING_BINDIR_NATIVE}/upx --best -q $exec
```

Comment:
UPX binary should be a variable
UPX ?= "${STAGING_BINDIR_NATIVE}/upx"
As well I would make the arguments configurable:
UPX_ARGS ?= "best -q"